tests/playbooks/container_repository.yaml

```diff
@@ -0,0 +1,148 @@
+    - name: Verify create repository
+      assert:
+        that:
+          - result.changed == true
+          - result.repository.name == 'test_container_repository'
+
+    - name: Create repository (2nd try)
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        state: present
+      register: result
+    - name: Verify create repository (2nd try)
+      assert:
+        that:
+          - result.changed == false
+
+    - name: Add description to repository
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        description: "repository created via ansible"
+        state: present
+      register: result
+    - name: Verify add description to repository
+      assert:
+        that:
+          - result.changed == true
+          - result.repository.description == "repository created via ansible"
+
+    - name: Add description to repository (2nd try)
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        description: "repository created via ansible"
+        state: present
+      register: result
+    - name: Verify add description to repository (2nd try)
+      assert:
+        that:
+          - result.changed == false
+
+    - name: Fake modify repository
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        state: present
+      register: result
+    - name: Verify fake modify repository
+      assert:
+        that:
+          - result.changed == false
+          - result.repository.description == "repository created via ansible"
+
+    - name: List repositories
+      pulp.squeezer.container_repository: {}
+      register: result
+    - name: Verify list repositories
+      assert:
+        that:
+          - result.changed == false
+          - result.repositories | length >= 1
+
+    - name: Read repository
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+      register: result
+    - name: Verify read repository
+      assert:
+        that:
+          - result.changed == false
+          - result.repository.name == 'test_container_repository'
+          - result.repository.description == "repository created via ansible"
+
+    - name: Remove description from repository
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        description: ""
+        state: present
+      register: result
+    - name: Verify remove description from repository
+      assert:
+        that:
+          - result.changed == true
+          - not result.repository.description
+
+    - name: Remove description from repository (2nd try)
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        description: ""
+        state: present
+      register: result
+    - name: Verify remove description from repository (2nd try)
+      assert:
+        that:
+          - result.changed == false
```

Comment:
This test may actually be superfluous. (Or let's say sufficient to be tested with file repos. It is mainly there, because with the python bindings for pulp it is impossible to remove the description.) OTOH it does not hurt. :shrug:

Reply:
I figured it was testing that it's possible to update an existing repository, both adding and removing a field?